src/service/client/mod.rs

```diff
@@ -25,15 +25,21 @@ impl crate::Service for Service {
 		let config = &args.server.config;
 		let resolver = args.require::<resolver::Service>("resolver");
 
+		let mut url_preview_builder = base(config)?
+			.dns_resolver(resolver.resolver.clone())
+			.redirect(redirect::Policy::limited(3));
+
+		#[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
```

Comment:
This is going to trigger the unnecessary mutable warning on macOS builds.

I don't see why this needs to be restricted to those 3 OS's, I would just remove the cfg attribute entirely. Especially since I don't see why this can't work on macOS either.

Reply:
This is restricted because it's restricted in upstream request - I think it's a Linux-exclusive feature.

https://docs.rs/reqwest/latest/src/reqwest/async_impl/client.rs.html#1393

Reply:
Should have fixed the mutability warning now.

Reply:
Oh I didn't know that, that's wack. I could absolutely have two ethernet connections on my Mac and send URL previews only through one of them 🥲 

And that's a lot better than the code I was about to suggest, thank you!